test/replication-luatest/persistent_gc_test.lua

```diff
@@ -0,0 +1,399 @@
+local t = require('luatest')
+local server = require('luatest.server')
+local cluster = require('luatest.replica_set')
+local fio = require('fio')
```

Comment:
Also that really strange error. Reproduced very rarely:

```
[009] replication-luatest/persistent_gc_test.lua                      [ fail ]
[009] Test failed! Output from reject file /tmp/t/rejects/replication-luatest/persistent_gc.reject:
[009] Tarantool version is 3.2.0-entrypoint-75-gecbc542e2a
[009] TAP version 13
[009] 1..7
[009] # Started on Thu Jun 13 14:22:17 2024
[009] # Starting group: Indirect replication of _gc_consumers
[009] ok     1	Indirect replication of _gc_consumers.test_indirect_replication_of_consumers
[009] # Starting group: Persistent gc
[009] ok     2	Persistent gc.test_gc_consumer_is_updated_by_relay
[009] ok     3	Persistent gc.test_gc_consumer_update_retry
[009] ok     4	Persistent gc.test_gc_consumer_is_not_updated_without_replica
[009] ok     5	Persistent gc.test_reconnect_inactive_consumer
[009] 
[009] [test-run server "luatest_server"] Last 15 lines of the log file /tmp/t/009_replication-luatest/persistent_gc_test.log:
[009] builtin/fio.lua:438: attempt to index local 'st' (a nil value)
[009] stack traceback:
[009] 	...g/tnt/tarantool/test-run/lib/luatest/luatest/capture.lua:144: in function <...g/tnt/tarantool/test-run/lib/luatest/luatest/capture.lua:138>
[009] 	[builtin#19]: at 0x55555590a2ac
[009] 	[C]: in function 'xpcall'
[009] 	...ing/tnt/tarantool/test-run/lib/luatest/luatest/utils.lua:38: in function 'run_tests'
[009] 	...ng/tnt/tarantool/test-run/lib/luatest/luatest/runner.lua:353: in function <...ng/tnt/tarantool/test-run/lib/luatest/luatest/runner.lua:348>
[009] 	[C]: in function 'xpcall'
[009] 	...tnt/tarantool/test-run/lib/luatest/luatest/capturing.lua:74: in function <...tnt/tarantool/test-run/lib/luatest/luatest/capturing.lua:72>
[009] 	[C]: in function 'xpcall'
[009] 	...ng/tnt/tarantool/test-run/lib/luatest/luatest/runner.lua:55: in function 'fn'
[009] 	...antool/test-run/lib/luatest/luatest/sandboxed_runner.lua:14: in function 'run'
[009] 	...arantool/test-run/lib/luatest/luatest/cli_entrypoint.lua:4: in function <...arantool/test-run/lib/luatest/luatest/cli_entrypoint.lua:3>
[009] 	...ogramming/tnt/tarantool/test-run/lib/luatest/bin/luatest:5: in main chunk
[009] Worker "009_replication-luatest" got failed test; restarted the server
```

Reply:
The reason is [the same](https://github.com/tarantool/tarantool/pull/9960#discussion_r1639762713): During traversal underlying xlogs were asynchronously deleted and `fio` didn't cope with it.